Add visit_try_cast to sqlalchemy compiler

dwolfeu opened the pull request:
<!-- Thank you for submitting a pull request! Find more information in our development guide at https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md and contact us on #python-client in Slack. -->
<!-- Provide an overview of the PR for maintainers and reviewers. -->
## Description
I have added a `visit_try_cast()` method to `trino.sqlalchemy.compiler.TrinoSQLCompiler`. This fixes [issue 473](https://github.com/trinodb/trino-python-client/issues/473).




<!-- Provide a user-friendly explanation, keep it brief if it isn't user-visible. -->
## Non-technical explanation
This allows Trino's [`try_cast()`](https://trino.io/docs/current/functions/conversion.html#try_cast) to be used in SQLAlchemy.



<!-- Mark the appropriate option with an (x). Propose a release note if you can. -->
## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

dwolfeu commented:
I just emailed the signed CLA.

hovaesco commented:
Thanks for your contribution! Could you please add a test for it?

dwolfeu commented:
Sure! I should add one [here](https://github.com/trinodb/trino-python-client/blob/master/tests/unit/sqlalchemy/test_compiler.py), right?

hovaesco commented:
> Sure! I should add one [here](https://github.com/trinodb/trino-python-client/blob/master/tests/unit/sqlalchemy/test_compiler.py), right?

Yes, correct.

dwolfeu commented:
> Thanks for your contribution! Could you please add a test for it?

I have added a unit test.

dwolfeu commented:
@hovaesco [My CLA has been processed.](https://github.com/trinodb/cla/commit/9d6a6272496df5e7fd426221b45bd563cd7fb6af)

hovaesco commented:
@cla-bot check

hashhar commented:
@dwolfeu Please make test conditional to run only with SQLAlchemy 2.x since it seems 1.x doesn't have `try_cast`? (or maybe it needs to be imported differently?).

dwolfeu commented:
> @dwolfeu Please make test conditional to run only with SQLAlchemy 2.x since it seems 1.x doesn't have `try_cast`? (or maybe it needs to be imported differently?).

@hashhar Thanks for reviewing this PR. I have added an exception to the unit test. Three tests still fail, but they seem to be unrelated to my changes.

dwolfeu commented:
@hashhar @hovaesco Would it be possible to resolve this PR by the weekend or is that being too impatient?

hashhar reviewed (approved):
I've squashed (logically a single change) and rebased on master (to fix CI).

Will merge, sorry for the slow work.

hashhar commented:
Fixed some linter errors.

dwolfeu commented:
> I've squashed (logically a single change) and rebased on master (to fix CI).
> 
> Will merge, sorry for the slow work.

Not at all! Thank you for the collaboration. :)
